test/e2e/apimachinery/garbage_collector.go

```diff
@@ -605,24 +609,33 @@ var _ = SIGDescribe("Garbage collector", func() {
 		if err := deployClient.Delete(deployment.ObjectMeta.Name, deleteOptions); err != nil {
 			framework.Failf("failed to delete the deployment: %v", err)
 		}
-		By("wait for 2 Minute to see if the garbage collector mistakenly deletes the rs")
-		err = wait.PollImmediate(5*time.Second, 2*time.Minute, func() (bool, error) {
-			return verifyRemainingDeploymentsReplicaSetsPods(f, clientSet, deployment, 0, 1, 2)
-		})
-		if err != nil {
-			err = fmt.Errorf("Failed to wait to see if the garbage collecter mistakenly deletes the rs: %v", err)
+		By("wait for 30 seconds to see if the garbage collector mistakenly deletes the rs")
```

Comment:
why poll at all? why not just sleep 30 seconds, then list and ensure length is 1? expecting to get a timeout error is confusing.

Reply:
@liggitt 
that is how I had it but changed it in favor of consistency with other similar code in this file
https://github.com/kubernetes/kubernetes/pull/60671#discussion_r172690853

Reply:
Would this be better? https://github.com/jennybuckley/kubernetes/commit/03b4b237c4c04172f587a24f184083e9e2a00f27#diff-4a7d8678f8505e86c1c6cc626cdc5eeeR613

Reply:
@jennybuckley your code is better, let's change the rest of code to comply with yours.